cmd/build.go

```diff
@@ -32,6 +32,6 @@ func NewBuildCmd(globalCfg *config.GlobalImpl) *cobra.Command {
 
 	f := cmd.Flags()
 	f.BoolVar(&buildOptions.EmbedValues, "embed-values", false, "Read all the values files for every release and embed into the output helmfile.yaml")
-
+	f.BoolVar(&buildOptions.DAG, "dag", false, "Outputs the directed acyclic graph of release dependencies, as defined by the \"needs:\" declarations in the helmfile.yaml")
```

Comment:
I think this should be a dedicated helmfile sub-command like `helmfile show-dag` as this doesn't actually build anything in the sense that helmfile-build does!

Also, the usage of this flag(and later the usage for the `show-dag` sub-command) should also explain how the output looks like.

For example...

```
It prints a table with 3 columns, GROUP, RELEASE, and DEPENDENCIES.

GROUP is the unsigned, monotonically increasing integer starting from 1. All the releases with the same GROUP are deployed concurrently. Everything in GROUP 2 starts being deployed only after everything in GROUP 1 got successfully deployed.

RELEASE is the release that belongs to the GROUP.

DEPENDENCIES is the list of releases that the RELEASE depends on. It should always be empty for releases in GROUP 1. DEPENDENCIES for a release in GROUP 2 should have some or all dependencies appeared in GROUP 1. It can be "some" because Helmfile simplifies the DAGs of releases into a DAG of groups, so that Helmfile always produce a single DAG for everything written in helmfile.yaml, even when there are technically two or more independent DAGs of releases in it.
```

Reply:
I've added a new `show-dag` command with the description above.
Also, a DAG-related changes from previous commits are removed from `build `command.

@mumoshu could you please check?